src/anomalib/models/image/glass/lightning_model.py

```diff
@@ -0,0 +1,328 @@
+            Determines the range for valid synthetic anomalies in the hypersphere or manifold.
+            Defaults to `0.75`.
+        random_selection_prob (float, optional): Probability used in random selection logic, such as anomaly mask
+          generation or augmentation choice.
+            Defaults to `0.5`.
+        learning_rate (float, optional): Learning rate for training the feature adaptor and discriminator networks.
+            Defaults to `0.0001`.
+        step (int, optional): Number of gradient ascent steps for anomaly synthesis.
+            Defaults to `20`.
+        svd (int, optional): Flag to enable SVD-based feature projection.
+            Defaults to `0`.
+    """
+
+    def __init__(
+        self,
+        input_shape: tuple[int, int] = (256, 256),
+        anomaly_source_path: str | None = None,
+        backbone: str = "resnet18",
+        pretrain_embed_dim: int = 1024,
+        target_embed_dim: int = 1024,
+        patchsize: int = 3,
+        patchstride: int = 1,
+        pre_trained: bool = True,
+        layers: list[str] | None = None,
+        pre_projection: int = 1,
+        discriminator_layers: int = 2,
+        discriminator_hidden: int = 1024,
+        discriminator_margin: float = 0.5,
+        pre_processor: PreProcessor | bool = True,
+        post_processor: PostProcessor | bool = True,
+        evaluator: Evaluator | bool = True,
+        visualizer: Visualizer | bool = True,
+        mining: int = 1,
+        noise: float = 0.015,
+        radius: float = 0.75,
+        random_selection_prob: float = 0.5,
+        learning_rate: float = 0.0001,
+        step: int = 20,
+        svd: int = 0,
+    ) -> None:
+        super().__init__(
+            pre_processor=pre_processor,
+            post_processor=post_processor,
+            evaluator=evaluator,
+            visualizer=visualizer,
+        )
+
+        if layers is None:
+            layers = ["layer1", "layer2", "layer3"]
+
+        self.model = GlassModel(
+            input_shape=input_shape,
+            anomaly_source_path=anomaly_source_path,
+            pretrain_embed_dim=pretrain_embed_dim,
+            target_embed_dim=target_embed_dim,
+            backbone=backbone,
+            pre_trained=pre_trained,
+            patchsize=patchsize,
+            patchstride=patchstride,
+            layers=layers,
+            pre_projection=pre_projection,
+            discriminator_layers=discriminator_layers,
+            discriminator_hidden=discriminator_hidden,
+            discriminator_margin=discriminator_margin,
+            step=step,
+            svd=svd,
+            mining=mining,
+            noise=noise,
+            radius=radius,
+            random_selection_prob=random_selection_prob,
+        )
+
+        self.learning_rate = learning_rate
+
+        if pre_projection > 0:
+            self.projection_opt = optim.AdamW(
```

Comment:
```suggestion
            self.projection_opt = optim.Adam(
```

This is what the official implementation says https://github.com/cqylunlun/GLASS/blob/6af03b9d7f7b33a1aebd69cd4c30a41bf020a2d1/glass.py#L99

Though in the paper they use Adam for both feature adapter and the discriminator

<img width="539" height="41" alt="Image" src="https://github.com/user-attachments/assets/322a5ec8-2d7d-4920-8e53-941363614341" />